src/directLine.ts

```diff
@@ -558,9 +564,12 @@ export class DirectLine implements IBotConnection {
                 } else {
                     return this.startConversation().do(conversation => {
                         this.conversationId = conversation.conversationId;
-                        this.token = this.secret || conversation.token;
+                        // Don't invoke acquireToken(), coulf fetch anew AAD token
+                        this.token = (this.acquireToken ? this.token : this.secret) || conversation.token;
```

Comment:
Why are we setting the secret as the token ever? This isn't a practice we should follow, right?

Reply:
Yeah, that doesn't look good. However, ABS API expects either a token or secret in the Authentication header. Maybe we should use a variable name that suits both.